meteor/client/ui/Prompter/PrompterView.tsx

```diff
@@ -543,27 +510,65 @@ export class PrompterViewInner extends MeteorReactComponent<Translated<IProps &
 		)
 	}
 }
-export const PrompterView = translateWithTracker<IProps, {}, ITrackedProps>(({ studioId }: IProps) => {
-	const studio = UIStudios.findOne(studioId)
-
-	const rundownPlaylist = RundownPlaylists.findOne(
-		{
-			activationId: { $exists: true },
-			studioId: studioId,
-		},
-		{
-			projection: {
-				trackedAbSessions: 0,
-				lastIncorrectPartPlaybackReported: 0,
-			},
-		}
-	) as Omit<DBRundownPlaylist, 'trackedAbSessions'> | undefined
+export function PrompterView(props: Readonly<IProps>): JSX.Element {
+	const i18nProps = useTranslation()
+
+	const studioIsReady = useSubscription(MeteorPubSub.uiStudio, props.studioId)
+	const playlistIsReady = useSubscription(MeteorPubSub.rundownPlaylistForStudio, props.studioId, true)
+
+	const playlist = useTracker(
+		() =>
+			RundownPlaylists.findOne(
+				{
+					studioId: props.studioId,
+					activationId: { $exists: true },
+				},
+				{
+					fields: {
+						_id: 1,
+					},
+				}
+			) as Pick<DBRundownPlaylist, '_id'> | undefined,
+		[props.studioId]
+	)
+	const rundownsIsReady = useSubscription(CorelibPubSub.rundownsInPlaylists, playlist ? [playlist._id] : [])
+
+	const [subsReady, setSubsReady] = useState(false)
+	useEffect(() => {
+		setSubsReady(studioIsReady && playlistIsReady && rundownsIsReady)
+	}, [studioIsReady, playlistIsReady, rundownsIsReady])
+
+	const studio = useTracker(() => UIStudios.findOne(props.studioId), [props.studioId])
 
-	return literal<ITrackedProps>({
-		rundownPlaylist,
-		studio,
-	})
-})(PrompterViewInner)
+	const rundownPlaylist = useTracker(
+		() =>
+			RundownPlaylists.findOne(
+				{
+					activationId: { $exists: true },
+					studioId: props.studioId,
+				},
+				{
+					projection: {
+						trackedAbSessions: 0,
+						lastIncorrectPartPlaybackReported: 0,
+					},
+				}
+			) as Omit<DBRundownPlaylist, 'trackedAbSessions'> | undefined,
+		[props.studioId]
+	)
+
+	return (
+		<PrompterViewInner
+			{...props}
+			t={i18nProps.t}
+			i18n={i18nProps.i18n}
```

Comment:
This seems weird, wouldn't it make more sense to provide `t`, `i18n`, `tReady` via a `withTranslation` HOC (i.e. replacing the `translateWithTracker` with a `withTranslation`?

Reply:
I think I did this because doing it with `withTranslation` requires doing that wrap in its own line, as the `PromperViewInner` is exported and types break if wrapped with `withTranslation`.
And this way avoid the extra wrapping component that `withTranslation` adds.

But I don't feel strongly that it should be done this way, and do agree that the manual props feels a little messy, so I've changed it to have the line `const PrompterViewContentWithTranslation = withTranslation()(PrompterViewContent)` which then gets used here